iteration/iteration_test.go

```diff
@@ -38,289 +38,273 @@ func (s *TestIterationRepository) TearDownTest() {
+		require.Nil(t, err)
+		assert.Equal(t, i.Name, res.Name)
+		assert.Equal(t, i.ID, res.ID)
+		expectedPath := i.Path.Convert()
+		require.NotNil(t, res)
+		assert.Equal(t, expectedPath, res.Path.Convert())
 	})
-	require.Nil(t, e)
 
-	its, err := repo.List(context.Background(), fxt.Spaces[0].ID)
-	assert.Nil(t, err)
-	assert.Len(t, its, 3)
-}
+	t.Run("list by space", func(t *testing.T) {
+		fxt := tf.NewTestFixture(s.T(), s.DB, tf.Spaces(2))
 
-func (s *TestIterationRepository) TestUpdateIteration() {
-	t := s.T()
-	resource.Require(t, resource.Database)
-	repo := iteration.NewIterationRepository(s.DB)
+		for i := 0; i < 3; i++ {
+			start := time.Now()
+			end := start.Add(time.Hour * (24 * 8 * 3))
+			name := "Sprint #2" + strconv.Itoa(i)
 
-	start := time.Now()
-	end := start.Add(time.Hour * (24 * 8 * 3))
+			i := iteration.Iteration{
+				Name:    name,
+				SpaceID: fxt.Spaces[0].ID,
+				StartAt: &start,
+				EndAt:   &end,
+			}
+			e := repo.Create(context.Background(), &i)
+			require.Nil(t, e)
+		}
+		// add iteration to the second space from our fixture
+		e := repo.Create(context.Background(), &iteration.Iteration{
+			Name:    "Other Spring #2",
+			SpaceID: fxt.Spaces[1].ID,
+		})
+		require.Nil(t, e)
 
-	fxt := tf.NewTestFixture(s.T(), s.DB,
-		tf.Iterations(1,
-			tf.UserActive(false),
-			func(fxt *tf.TestFixture, idx int) error {
+		its, err := repo.List(context.Background(), fxt.Spaces[0].ID)
+		assert.Nil(t, err)
+		assert.Len(t, its, 3)
+	})
+
+	t.Run("load children", func(t *testing.T) {
+		fxt := tf.NewTestFixture(s.T(), s.DB,
+			tf.Iterations(3, func(fxt *tf.TestFixture, idx int) error {
 				i := fxt.Iterations[idx]
-				i.Name = "Sprint #24"
-				i.StartAt = &start
-				i.EndAt = &end
+				switch idx {
+				case 0:
+					i.Name = "Top level iteration"
+				case 1:
+					i.Name = "Level 1 iteration"
+					i.MakeChildOf(*fxt.Iterations[idx-1])
+				case 2:
+					i.Name = "Level 2 iteration"
+					i.MakeChildOf(*fxt.Iterations[idx-1])
+				}
 				return nil
-			},
-		),
-	)
-
-	i := *fxt.Iterations[0]
-
-	desc := "Updated item"
-	i.Description = &desc
-	updatedName := "Sprint 25"
-	i.Name = updatedName
-	// update iteration with new values of Name and Desc
-	updatedIteration, err := repo.Save(context.Background(), i)
-	require.Nil(t, err)
-	assert.Equal(t, updatedIteration.Name, updatedName)
-	assert.Equal(t, *updatedIteration.Description, desc)
-
-	changedStart := start.Add(time.Hour)
-	i.StartAt = &changedStart
-	changedEnd := start.Add(time.Hour * 2)
-	i.EndAt = &changedEnd
-	// update iteration with new values of StartAt, EndAt
-	updatedIteration, err = repo.Save(context.Background(), i)
-	require.Nil(t, err)
-	assert.Equal(t, changedStart, *updatedIteration.StartAt)
-	assert.Equal(t, changedEnd, *updatedIteration.EndAt)
+			}),
+		)
+		i1 := *fxt.Iterations[0]
+		i2 := *fxt.Iterations[1]
+		i3 := *fxt.Iterations[2]
+
+		// fetch all children of top level iteraiton
```

Comment:
minor typo in comment: `iteraiton`

Reply:
Thanks, done in https://github.com/fabric8-services/fabric8-wit/pull/1719/commits/2949ad3902c40f46bca32e444c41e2c67eb4f458